### ISSUE TITLE
[Feature Request] Ability to turn off commenting on pull requests

### ISSUE THREAD

**Issue by anton-buyskikh** [5monkeys/cobertura-action#76]
It would be nice to have a flag such as `post_on_pr: true/false`, which would toggle this option on/off.
The report should be generated in either case and be accessible via the Actions tab, but not posted to the PR discussion thread if desired.

Reasons:
- These comments draw unnecessary attention
- In case of matrix strategies, multiple comments are posted

**Comment by tibbing** [comment]
Agreed, this would also fix https://github.com/5monkeys/cobertura-action/issues/40

**Comment by rjulius23** [comment]
Normally, if you do not specify teh pullRequestNumber parameter it should skip the comment, however I think the issue is with this part:

`
  const changedFiles = onlyChangedFiles
    ? await listChangedFiles(pullRequestNumber)
    : null;

  const reports = await processCoverage(path, { skipCovered });
  const comment = markdownReport(reports, commit, {
    minimumCoverage,
    showLine,
    showBranch,
    showClassNames,
    showMissing,
    showMissingMaxLength,
    linkMissingLines,
    linkMissingLinesSourceDir,
    filteredFiles: changedFiles,
    reportName,
  });
`

This is the section in the action.js from line 54 to 70.  This runs even if the pullRequestNumber is null, and that fails the run with Not Found error. Any reason the above part is not inside the if (pullRequestNumber) block at line 76 ?

Cheers,

**Comment by algorys** [comment]
Is there a chance to have this fix please ? 🙏 @zymotik ?

Because today a simple push to main didn't work and this plugin is great !

**Comment by tcs-ujjawal** [comment]
> It would be nice to have a flag such as `post_on_pr: true/false`, which would toggle this option on/off. The report should be generated in either case and be accessible via the Actions tab, but not posted to the PR discussion thread if desired.
> 
> Reasons:
> 
> * These comments draw unnecessary attention
> * In case of matrix strategies, multiple comments are posted

Looks like no one is looking into this issue !!!
